app/storage/s3ResourceHandler.js

```diff
@@ -19,91 +19,113 @@
 const conf = require('./../conf.js').conf;
 const S3ClientClass = require('./s3NewClient');
 const cipher = require('./cipher');
+const _ = require('lodash');
 
 class S3ResourceHandler {
 
-  constructor(logger, resourceKey, bucketName, location, endpoint) {
+  constructor(args) {
+    let { logger, path, bucketName=null, bucketConfObj=null, location, endpoint, org } = args;
+    if(!bucketConfObj && !bucketName){
+      throw new Error('Pass a bucketConfObj or bucketName');
+    }
```

Comment:
GIT's diff is a bit garbled, but the original code was:
```
if (!resourceKey || !bucketName) {
      throw new Error(`Path (${resourceKey}) and/or bucket name (${bucketName}) is not specified`);
}
```
By changing it to `&&` it changes the behavior and makes it confusing.  E.g. why would it be necessary to construct an s3ResourceHandler without specifying the bucketName and have it default to using the common (not per-org, not encrypted) `resources` bucket?  That wasn't possible before.